pkg/controller/volume/attachdetach/testing/testvolumespec.go

```diff
@@ -294,6 +294,10 @@ func (plugin *TestPlugin) NewAttacher() (volume.Attacher, error) {
 	return &attacher, nil
 }
 
+func (plugin *TestPlugin) NewDeviceMounter() (volume.DeviceMounter, error) {
+	return plugin.NewAttacher()
```

Comment:
Hm, I wonder if this would cause any issues for attachable plugins if they had been keeping some state in the Attacher object.

Reply:
I'm worried about situations where a plugin may have kept some state to serialize between WaitForAttach and MountDevice, like a lock or some cached data.

Reply:
Got it, for now, i can not find the state kept between WaitForAttach and MountDevice through the plugins.
But should we cache the attacher info for the situations you are worried about ?

/cc @jsafrane @saad-ali

Reply:
I went through volume plugins, none of them really passes information from WaitForAttach to MountDevice via the attacher instance.